src/template-shadowroot.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+let hasNative: boolean|undefined;
+export function hasNativeDeclarativeShadowRoots(): boolean {
+  if (hasNative === undefined) {
+    const div = document.createElement('div');
```

Comment:
It looks like `HTMLTemplateElement` is intended to have a `shadowRoot` property:

```js
function supportsDeclarativeShadowDOM() {
  return 'shadowRoot' in HTMLTemplateElement.prototype;
}
```

from https://github.com/mfreed7/declarative-shadow-dom/blob/master/README.md#feature-detection-and-polyfilling

Is that the case in Canary?

Reply:
Awesome 👍